modules/stanford_earth_capx/src/EarthCapxInfo.php

```diff
@@ -0,0 +1,249 @@
+        }
+      }
+    }
+  }
+
+  /**
+   * get the photo timestamp from the cap url
+   */
+  private function getPhotoTimestamp($photoUrl = "") {
+    $photo_ts = '';
+    if (!empty($photoUrl) && is_string($photoUrl)) {
+      $ts1 = strpos($photoUrl,"ts=");
+      if ($ts1 !== false) {
+        $ts2 = substr($photoUrl,$ts1);
+        if (strpos($ts2,"&") !== false)  {
+          $photo_ts = substr($ts2,3,strpos($ts2,"&")-3);
+        } else {
+          $photo_ts = substr($ts2,3);
+        }
+      }
+    }
+    return $photo_ts;
+  }
+  
+  /**
+   * see if we have an existing and current profile photo_id
+   * if we do, return its fid. if we don't, return false
+   * @param array $source
+   */
+  public function currentProfilePhotoId($source = []) {
+    if (empty($source['profile_photo']) || empty($this->profilePhotoFid)
+      || empty($this->profilePhotoTimestamp)) {
+      return false;
+    }
+    $photo_ts = $this->getPhotoTimestamp($source['profile_photo']);
+    if ($photo_ts == $this->profilePhotoTimestamp) {
+      return $this->profilePhotoFid;
+    } else {
+      return false;
+    }
+  }
+  
+  /**
+   * return true if profile should be updated because it is new
+   * or because the etag has changed.
+   * @param array $source
+   */
+  public function getOkayToUpdateProfile($source = [], $photoId = 0) {
+    $oktoupdate = false;
+    $msg = new MigrateMessage();
+    if (empty($source['sunetid']) ||
+      $this->status == self::EARTH_CAPX_INFO_INVALID ||
+      $source['sunetid'] !== $this->sunetid) {
+        $msg->display('Unable to validate new profile information.', 'error');
+    } else if ($this->status == self::EARTH_CAPX_INFO_NEW) {
+      $oktoupdate = true;
+    } else if ($this->status == self::EARTH_CAPX_INFO_FOUND) {
+      $source_etag = '';
+      if (!empty($source['etag'])) {
+        $source_etag = $source['etag'];
+      }
+      if ($this->etag !== $source_etag || $this->profilePhotoFid !== $photoId) {
+        $oktoupdate = true;
+      }
+    }
+    return $oktoupdate;
+  }
+  
+  /**
+   * update the table with information from the source array and destination id
+   * only do the operation if information has changed
+   * 
+   * @param array $source
+   */
+  public function setInfoRecord($source = [], $entity_id = 0, $photo_id = 0) {
+    // see if we have valid sunetids
+    $msg = new MigrateMessage();
+    if (empty($source['sunetid']) ||
+      $this->status == self::EARTH_CAPX_INFO_INVALID) {
+        $msg->display('Unable to update EarthCapxInfo table. Missing source id.','error');
+        return;
+    }
+    if ($source['sunetid'] !== $this->sunetid) {
+      $msg->display('Unable to update EarthCapxInfo table. Mismatched ids: ' .
+        $source['sunetid'] . ', ' . $this->sunetid);
+      return;
+    }
+
+    // get the fields from the source array
+    $source_etag = '';
+    if (!empty($source['etag'])) $source_etag = $source['etag'];
```

Comment:
Short php format not allowed in Drupal coding standards.

Reply:
Fixed.